rules/ml/persistence_ml_rare_process_by_host_windows.toml

```diff
@@ -26,38 +26,83 @@ machine_learning_job_id = ["v3_rare_process_by_host_windows"]
+  - Investigate the process metadata, like the digital signature, directory, etc., to obtain more context that can
+  indicate whether the executable is associated with an expected software vendor or package.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Consider the user as identified by the `user.name` field. Is this program part of an expected workflow for the user
+who ran this program on this host?
+- Validate the activity is not related to planned patches, updates, network administrator activity, or legitimate
+software installations.
+- Validate if the activity has a consistent cadence (for example, if it runs monthly or quarterly), as it could be part
+of a monthly or quarterly business process.
+- Examine arguments and working directory. These may provide indications as to the source of the program or the nature
+of the tasks it is performing.
+- Examine the host for derived artifacts that indicates suspicious activities:
+  - Analyze the process executable using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by
+      filtering by the process' `process.entity_id`.
+      - Examine the DNS cache for suspicious or anomalous entries.
+        - !{osquery{"query":"SELECT * FROM dns_cache", "label":"Osquery - Retrieve DNS Cache"}}
+    - Use the Elastic Defend registry events to examine registry keys accessed, modified, or created by the related
+    processes in the process tree.
+    - Examine the host services for suspicious or anomalous entries.
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services","label":"Osquery - Retrieve All Services"}}
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE \"%LocalSystem\" OR user_account LIKE \"%LocalService\" OR user_account LIKE \"%NetworkService\" OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != \"trusted\"","label":"Retrieve Service Unisgned Executables with Virustotal Link"}}
+  - Retrieve the files' SHA-256 hash values using the PowerShell `Get-FileHash` cmdlet and search for the existence and
+  reputation of the hashes in resources like VirusTotal, Hybrid-Analysis, CISCO Talos, Any.run, etc.
```

Comment:
The nested list levels here are pretty complex and hard to follow. Would it be possible to have just two or three levels at most? It seems like most of the actions would OK on the same level:

- Examine the host for artifacts:
  - Analyze the process executable
  - Observe and collect
  - Use the ED network events
  - Examine the DNS cache
    - osquery example
  - Use the ED registry events
  - Examine the host services
    - osquery example
    - osquery example
    - osquery example
  - Retrieve the files' SHA-256

Reply:
I'll keep this noted, but as those are implemented in other Rules that are not involved in this PR, so I'll not change the format by now